service/src/main/resources/db/changesets/20220301_datasettable.yaml

```diff
@@ -0,0 +1,40 @@
+databaseChangeLog:
+  - changeSet:
+      id: dataset_table
+      author: sehsan
+      changes:
+        - createTable:
+            tableName: dataset
+            columns:
+              - column:
+                  name: id
+                  type: ${uuid_type}
+                  defaultValueComputed: ${uuid_function}
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: dataset_id
+                  type: text
+                  constraints:
+                    nullable: false
+              - column:
+                  name: storage_system
+                  type: text
+                  constraints:
+                    nullable: false
+              - column:
+                  name: metadata
+                  type: jsonb
+                  constraints:
+                    nullable: true
+              - column:
+                  name: created_date
+                  type: timestamptz
+                  defaultValueComputed: now()
+                  constraints:
+                    nullable: false
+        - addUniqueConstraint:
```

Comment:
```suggestion
        # ensure there can only be one dataset_id, storage_system combination
        - addUniqueConstraint:
```

Reply:
I missed this at first, maybe this comment is redundant though 👍

Reply:
I'll add it! It's not really clear from just looking at the yaml if each field needs to be unique or the combination of the two.